model/task_json.go

```diff
@@ -53,6 +63,25 @@ var (
 	TaskJSONTagKey                 = bsonutil.MustHaveTag(TaskJSON{}, "Tag")
 )
 
+// Returns distinct list of all tags for given `projectId`
+func GetDistinctTagNames(projectId string) ([]Tag, error) {
+	out := []Tag{}
+
+	return out, db.Aggregate(TaskJSONCollection, []bson.M{
```

Comment:
can you separate this into a few steps so that you can annotate the error and explicitly make sure that the out value is nil in error cases?

Reply:
Do you mean?
```
out, err = db.Aggregate(...)
return out, err
```
I do not have any specific error handling logic.

Reply:
I mean't 

```
out, err := db.Aggregate(...)
if err != nil {
return nil, errors.Wrap(err, "problem with tag query") 
}
return out, nil 
```